### ISSUE TITLE
Will give more detail about the network?

### ISSUE THREAD

**Issue by Windaway** [wukaoliu/CVPR2020-HAttMatting#9]
I have reimplement the paper. How ever my implement cost a lot of memory with a just resnet50 backbone, which is very strange. The paper said one P100 can hold a  batchsize of 4. Can you give more detail? Such as output stride of the backbone, and the channel number of the ASPP.

By analyzing the Diagram, did resnext101 remove the maxpool layer? And use the strides [1,2,2,1] and dilations[1, 1, 1, 2] like os=16 (delete maxpool to still make output stride = 8)?

**Comment by Twice22** [comment]
Hey!

I have myself reimplemented the whole paper with some assumptions regarding some parts of the Neural Network.
The code is running and actually the network is learning but then entered into failure mode with a discriminator loss of 0.

If you still work on it, maybe we could help each other ^^.

Also, for your information, I can train with batch-size of 2 on a 11GB single gpu :)

**Comment by aricsong1995** [comment]
@Twice22 Hey! i'm new and i'm also working on reimplementing the network and got stucked. Would you mind share your network modle (just the network) with me please? Thanks a lot!

**Comment by Twice22** [comment]
Well, I can help you ofc but you need to show me first that you worked on it ^^

**Comment by aricsong1995** [comment]
Hi Victor, Thanks a lot for replying to me!  As I may have mentioned, I'm
still a newbee to python and pytorch.This is my first time trying to
re-implement a complex network. And it's really challenging for me to be
honest, So, this is what I have done for the last two weeks(almost):

*Reading*:

*ResNext  Basebone*:
Deep Residual Learning for Image Recognition
Aggregated Residual Transformations for Deep Neural Networks
*ASAP*:
Deeplab: Semantic image segmentation with deep convolutional nets, atrous
convolution, and fully connected crfs.
*Spatial and Channel wise Attention:*
  SCA-CNN: Spatial and Channel-wise Attention in Convolutional Networks for
Image Captioning
*PatchGan*:
Image-to-Image Translation with Conditional Adversarial Nets
Cycle-Gan

*Programming:*
Since I spent lots of time reading, I felt overwhelmed by all those new
concepts and information. So  my friend suggested that I start to write and
test some code for each module and then try to put them together. So from
this week, I start to  explore each module, I also have attached a part of
the code just to show that I'm actually learning each module of
HAttmatting.I'm using  tensorboard to visualize each module(that i found on
github) for me to get better understanding.
I also found that the true understanding must be at the code, not the
paper!
Anyway!The biggest problem I encounter is that I still can't find any code
reference for the HattMatting network, which I found frustrating to get
going.Not to even mention that the paper did not cover all the network
details. Since I'm new to all module concepts, it's really hard for me to
put everything together.
I guess it's always hard to do it the first time. But if I can successfully
implement the network, I will definitely share it back with you as well or
on github!

Victor, Thanks again if you could help! And If you could, it would be great
if you can share the code of the network model(pytorch) with me. I;m going
to read your code carefully first and then try to build mine.

Best Regards,
Aric
